tools/export/makefile/__init__.py

```diff
@@ -54,7 +54,8 @@ class Makefile(Exporter):
         "MCU_NRF51Code.binary_hook",
         "TEENSY3_1Code.binary_hook",
         "LPCTargetCode.lpc_patch",
-        "LPC4088Code.binary_hook"
+        "LPC4088Code.binary_hook",
+        "PSOC6Code.complete"
```

Comment:
Considering that this adds a checksum to the binary, does the device boot without this checksum? If not, please remove any exporters that can't/don't edit this checksum value, including make.

Reply:
Actually, this checksum is only used/verified by Cypress programming tools (PSoC Programmer, KitProg2). Our new, DAPLink based interface simply ignores it, so there is no longer strict requirement to have a checksum calculated at build time . Nevertheless, I've decided to retain these calculations to have a final image compatible with a broader range of development tools. 
To answer your question - exported projects do work, albeit with limitations similar to bootloader case. One would usually have to resort to some other means of uploading required M0 image (except for makefile project, which can build combined image using srec_cat tool as for the Nordic chips).
As our interface does not perform 'whole chip erase' by default, it is possible to copy M0 and M4 images separately. Also e.g. IAR does not perform chip erase, so development in IAR IDE is possible provided the chip is first uploaded with M0 image.

Reply:
I appreciate the explanation. This is good to stay then!